### PR TITLE
For #4201 fix app crash on renaming a collection 

### PR DESCRIPTION
**Pull request by nikit19**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

App crashed because fragment was not attached to context

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/collections/CreateCollectionFragment.kt [hunk 1]
```diff
@@ -151,7 +151,7 @@ class CreateCollectionFragment : DialogFragment() {
                     dismiss()
                     viewLifecycleOwner.lifecycleScope.launch(Dispatchers.IO) {
                         requireComponents.core.tabCollectionStorage.renameCollection(it.collection, it.name)
```
Comment (st3fan) [inline app/src/main/java/org/mozilla/fenix/collections/CreateCollectionFragment.kt]: To be consistent with the change for `requireComponents.analytics`, should this become `context?.components?.core`?

**Comment by st3fan** [comment]
@boek would this be something we uplift to an `rc.2` build?

File: app/src/main/java/org/mozilla/fenix/collections/CreateCollectionFragment.kt (hunk 1 above)
Comment (colintheshots) [reply app/src/main/java/org/mozilla/fenix/collections/CreateCollectionFragment.kt]: Yes, I agree. This definitely seems required if the latter is crashing.

**Comment by nikit19** [comment]
PR Updated

**Comment by boek** [comment]
@st3fan yes :)

**Comment by colintheshots** [comment]
The TaskCluster build failed due to a 500 error from Github. Re-running.

**Pull request merged**